client/src/components/Sidebar/style.scss

```diff
@@ -0,0 +1,88 @@
+$transition-time: 0.3s;
+$primary: #FC8030;
+$background: #000000BA;
+$white: #FCFCFC;
+$spacing: 0.5rem;
+$buttons-margin: 0.25rem;
+
+
+button {
+  padding: $spacing;
+  border-radius: $spacing;
+  color: white;
+  transition: $transition-time;
+  margin: 0 $buttons-margin;
+  cursor: pointer;
+}
+
+button[data-active=active] {
+  background-color: $primary;
+
+  &:hover {
+  }
+}
+
+button[data-active=inactive] {
+  background-color: transparent;
+
+  &:hover {
+    background-color: #ffffff22;
+  }
+}
+
+.sidebar {
+  position: absolute;
+  z-index: 100;
+  top: $spacing;
+  right: $spacing;
+  background-color: $background;
+  color: $white;
+  transition: width $transition-time, height $transition-time, transform $transition-time;
+
+  // TODO: extract to glass morphism
+  backdrop-filter: blur( 4px );
+  -webkit-backdrop-filter: blur( 4px );
+
+  &__head {
+    display: flex;
+    justify-content: flex-end;
+    margin: $buttons-margin;
+  }
+
+  &__body {
+    padding: $spacing;
+  }
+
+  &__userlist-item {
+    display: flex;
+    align-items: center;
+    // I'm just putting number that looks good
+    margin: calc($spacing * 1.5) $spacing;
+
+    span {
+      margin-left: $spacing;
+    }
+  }
+
+}
+
+.sidebar[data-open=open] {
+  border-radius: 16px;
+  height: calc(100vh - 2 * $spacing);
```

Comment:
Was it not possible to use flexbox for the sidebar so that you can avoid using calc?
I personally like to avoid using calc() as much as possible since it's hard to maintain.

Reply:
even with flex, still need to subtract the margin somehow and 100% would not work since it would shrink to however size it is. I think trying to solve that would lead to even worse maintainability.

Reply:
That makes sense thanks for looking into it